### PR TITLE
feat(filters): Hide waiting list filters

### PR DESCRIPTION
**Pull request by lancetipton**
**Ticket**: [ZEN-491](https://jira.simpleviewtools.com/browse/ZEN-491)

## Context
* `Waiting list` and `on waiting list` labels should only be shown if one or more of the viewable sessions has **isWaitingListAvailable** is set to **true**
* Viewable sessions refer to the filtered sessions based on the attendees restriction

## Goal
* Only show the waiting list filter labels when at least one sessions has isWaitingListAvailable set to true

## Updates
* `src/actions/session/filters/applySessionFilters.js`
  * Added check to see if any sessions have a waiting list
* `src/actions/session/filters/setWaitingListFilters.js`
  * Added key to store, to save the waiting list active state
* `src/actions/session/mapSessionInterface.js`
  * Updated mapSessionInterface to call applySessionFilters
    * This way it's not calling setAgendaSessions twice on when initializing
* `src/components/modals/filter.js`
  * Updated filter modal to show waiting list labels based on it's state
* `src/hooks/sessions/useWaitingListActive.js`
  * Added hook to pull waiting list state from the store
* `src/components/sessions.js`
  * Added a small update to call `useStyle` instead of theme.get
* `src/mocks/eventsforce/bookingStates/waitingList.js`
  * Added test-data options to test no waiting list lable state

## Testing
**Test 1**
* Run the evf package => `keg evf pack run evf:zen-491-filter-modal-labels`
  * Open you browser to http://evf-zen-491-filter-modal-labels.local.kegdev.xyz/
    * Open the filters modal, and ensure `Waiting List` and `On Waiting List` are displayed
  * In the test-data editor, set the booking state to be `waitingList-single-noWaitingListLabel`, and hit apply
    * It's near the end of the list of booking state items
    * Open the filter modal and ensure the `Waiting List` and `On Waiting List` are **NOT** displayed
  * In the test-data editor, set the booking state to be `waitingList-group-noWaitingListLabel`, and hit apply
    * It's near the end of the list of booking state items
    * Open the filter modal and ensure the `Waiting List` and `On Waiting List` are **NOT** displayed

**Test 2**
* Run the consumer package => `keg consumer pack run consumer:zen-491-filter-modal-labels`
  * **NOTE**
    * The test-data have been updated to allow testing when no sessions has `isWaitingListAvailable` set `true`
    * You can **NOT** test other states in the test consumer
  * Open you browser to http://consumer-zen-491-filter-modal-labels.local.kegdev.xyz/
  * Open the filters modal, and ensure `Waiting List` and `On Waiting List` are **NOT** displayed

### CODE REVIEW

**Review by mpcarolin — CHANGES_REQUESTED** [review]
Looks good and passed testing.

I left 3 small suggestions. Let me know what you think. 

I also left a long comment about the derived state, but it's not blocking right now. I just wanted to get all my thoughts out, and even if we were to do what suggest, I think it would deserve its own ticket.

**Review by mpcarolin — APPROVED** [review]
Looks good. 

Approved, although if you have time, please also add a jsdoc to `applySessionsFilters`

File: src/actions/session/filters/setWaitingListFilters.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  const waitingListActive = sessions.some(session => get(session, searchPath))
+
+  dispatch({
+    type: ActionTypes.UPSERT_ITEM,
+    payload: {
+      item: waitingListActive,
+      category: CATEGORIES.SETTINGS,
+      key: SUB_CATEGORIES.WAITING_LIST_ACTIVE,
+    },
+  })
```
[the remark this answers is not in the document]
Comment (mpcarolin) [reply src/actions/session/filters/setWaitingListFilters.js]: Spoke about this on call - we decided we would tackle this at some point in the future, perhaps with the visitapps work.

File: src/actions/session/filters/applySessionFilters.js [hunk 2]
```diff
@@ -11,11 +12,12 @@ const { CATEGORIES, SUB_CATEGORIES } = Values
 /**
  * applyFilters
  */
```
[the remark this answers is not in the document]
Comment (lancetipton) [reply src/actions/session/filters/applySessionFilters.js:18]: updated

File: src/actions/session/filters/applySessionFilters.js [hunk 3]
```diff
@@ -11,11 +12,12 @@ const { CATEGORIES, SUB_CATEGORIES } = Values
...
-export const applySessionFilters = () => {
+export const applySessionFilters = (sessions, agendaDays) => {
   const { items } = getStore()?.getState()
   const selectedFilters = items?.filters?.selectedFilters || []
-  const sessions = items?.sessions
-  const agendaDays = items?.agendaDays
+
+  sessions = sessions || items?.sessions
+  agendaDays = agendaDays || items?.agendaDays
```
[the remark this answers is not in the document]
Comment (lancetipton) [reply src/actions/session/filters/applySessionFilters.js:24]: updated

File: src/actions/session/filters/setWaitingListFilters.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+const { CATEGORIES, SUB_CATEGORIES } = Values
+
+/**
+ * Checks if any sessions have a waiting list available
+ * Then updates the store base on the result
+ * @param {Array.<import('SVModels/session').Session>} sessions
+ *
+ */
+export const setWaitingListFilters = sessions => {
```
[the remark this answers is not in the document]
Comment (lancetipton) [reply src/actions/session/filters/setWaitingListFilters.js]: updated

**Pull request merged**